core/modules/filters/fuzzy.js

```diff
@@ -0,0 +1,237 @@
+		fields.push("title");
+		fields.push("tags");
+		fields.push("text");
+	}
+
+	// Dice coefficient for string similarity
+	function diceCoefficient(a, b) {
+		if (a === b) return 1;
+
+		const bigrams = new Map();
+		for (let i = 0; i < a.length - 1; i++) {
+			const bigram = a.slice(i, i + 2);
+			bigrams.set(bigram, (bigrams.get(bigram) || 0) + 1);
+		}
+		let intersectionSize = 0;
+		for (let i = 0; i < b.length - 1; i++) {
+			const bigram = b.slice(i, i + 2);
+			const count = bigrams.get(bigram) || 0;
+			if (count > 0) {
+				bigrams.set(bigram, count - 1);
+				intersectionSize++;
+			}
+		}
+		return (2.0 * intersectionSize) / (a.length + b.length);
+	}
+
+	// Our scoring function
+	function scoreMatch(term, haystack, matchPosition) {
+		// const substr = haystack.substr(matchPosition, term.length);
+		// If we compare against the whole field content, shorter results are scored better
+		return diceCoefficient(term, haystack);
+	};
+
+	// Function to check a given tiddler for the search term
+	function searchTiddler(title) {
+		var tiddler = $tw.wiki.getTiddler(title);
+		if(!tiddler) {
+			tiddler = new $tw.Tiddler({title: title, text: "", type: "text/vnd.tiddlywiki"});
+		}
+		var contentTypeInfo = $tw.config.contentTypeInfo[tiddler.fields.type] || $tw.config.contentTypeInfo["text/vnd.tiddlywiki"],
+			searchFields;
+		// Get the list of fields we're searching
+		if(options.excludeField) {
+			searchFields = Object.keys(tiddler.fields);
+			$tw.utils.each(fields,function(fieldName) {
+				var p = searchFields.indexOf(fieldName);
+				if(p !== -1) {
+					searchFields.splice(p,1);
+				}
+			});
+		} else {
+			searchFields = fields;
+		}
+
+		if(!terms) {
+			return true;
+		}
+
+		var fuzzyMatched = new Array(terms.length).fill(false); // Track each term separately for words mode
+		var bestScore = new Array(terms.length).fill(0); // Initialize with zero score for each search term
+		for(var termIndex=0; termIndex<terms.length; termIndex++) { // iterate over all terms
+			for(var fieldIndex=0; fieldIndex<searchFields.length; fieldIndex++) { // check which field has the best match score if there are multiple fields with matches
+				var fieldName = searchFields[fieldIndex];
+				// Don't search the text field if content type is binary
+				if(fieldName === "text" && contentTypeInfo.encoding !== "utf8") {
+					break;
+				}
+				var str = tiddler.fields[fieldName];
+				if(str) {
+					if($tw.utils.isArray(str)) {
+						// If the field value is an array, test each regexp against each field array entry and fail if each regexp doesn't match at least one field array entry
+						str = str.slice(); // make a copy for lowercase mutation
+						for(var s=0; s<str.length; s++) {
+							str[s] = str[s].toLowerCase();
+							const matchPosition = dmpObject.match_main(str[s], terms[termIndex], 0);
+							if(matchPosition !== -1) {
+								const score = scoreMatch(terms[termIndex], str[s], matchPosition);
```

Comment:
IMO `score` and `matchPosition` variable definition should be moved above the outer `for`, and it should be `var`. IMO const does not make sense here.